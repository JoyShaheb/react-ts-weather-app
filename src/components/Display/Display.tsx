import { FC } from "react";
import Card from "../Card/Card";
import { Grid } from "@mui/material";
import AirIcon from "@mui/icons-material/Air";
import OpacityIcon from "@mui/icons-material/Opacity";
import WaterIcon from "@mui/icons-material/Water";
import WbSunnyIcon from "@mui/icons-material/WbSunny";
import WbTwilightIcon from "@mui/icons-material/WbTwilight";
import { iResult } from "../../types";
import TimeZone from "../TimeZone/TimeZone";

interface iDisplay {
  data: iResult;
}

const Display: FC<iDisplay> = ({ data }) => {
  const { main, wind, sys, timezone } = data?.result || {};

  return (
    <Grid container spacing={1}>
      <Grid item xs={6}>
        <Card
          icon={<OpacityIcon fontSize="medium" />}
          label="Temperature"
          data={main?.temp}
          unit="Â°C"
        />
      </Grid>
      <Grid item xs={6}>
        <Card
          icon={<AirIcon fontSize="medium" />}
          label="Wind"
          data={wind?.speed}
          unit="km/h"
        />
      </Grid>
      <Grid item xs={6}>
        <Card
          icon={<OpacityIcon fontSize="medium" />}
          label="Humidity"
          data={main?.humidity}
          unit="%"
        />
      </Grid>
      <Grid item xs={6}>
        <Card
          icon={<WaterIcon fontSize="medium" />}
          label="Pressure"
          data={main?.pressure}
          unit="hPa"
        />
      </Grid>
      <Grid item xs={6}>
        <Card
          icon={<WbSunnyIcon fontSize="medium" />}
          label="Sun Rise"
          // @ts-ignore
          data={TimeZone({ input: sys?.sunrise, timezone })}
        />
      </Grid>
      <Grid item xs={6}>
        <Card
          icon={<WbTwilightIcon fontSize="medium" />}
          label="Sun Set"
          // @ts-ignore
          data={TimeZone({ input: sys?.sunset, timezone })}
        />
      </Grid>
    </Grid>
  );
};

export default Display;
