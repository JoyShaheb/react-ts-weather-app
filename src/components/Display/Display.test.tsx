import React from "react";
import { render } from "@testing-library/react";
import Display from "./Display";

describe("Display component", () => {
  it("should match snapshot", () => {
    const data = {
      result: {
        coord: {
          lon: 90.4074,
          lat: 23.7104,
        },
        weather: [
          {
            id: 701,
            main: "Mist",
            description: "mist",
            icon: "50n",
          },
        ],
        base: "stations",
        main: {
          temp: 20.99,
          feels_like: 21.18,
          temp_min: 20.99,
          temp_max: 20.99,
          pressure: 1013,
          humidity: 78,
        },
        visibility: 1800,
        wind: {
          speed: 0,
          deg: 0,
          gust: 0,
        },
        clouds: {
          all: 75,
        },
        dt: 1671730915,
        sys: {
          type: 1,
          id: 9145,
          country: "BD",
          sunrise: 1671669396,
          sunset: 1671707810,
        },
        timezone: 21600,
        id: 1185241,
        name: "Dhaka",
        cod: 200,
      },
      error: false,
      isLoading: false,
    };
    

    const { container } = render(<Display data={data} />);
    expect(container).toMatchSnapshot();
  });
});
