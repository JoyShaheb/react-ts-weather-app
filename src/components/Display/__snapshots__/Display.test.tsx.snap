// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        class="MuiStack-root Card css-1sy57ly-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="OpacityIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.66 8 12 2.35 6.34 8C4.78 9.56 4 11.64 4 13.64s.78 4.11 2.34 5.67 3.61 2.35 5.66 2.35 4.1-.79 5.66-2.35S20 15.64 20 13.64 19.22 9.56 17.66 8zM6 14c.01-2 .62-3.27 1.76-4.4L12 5.27l4.24 4.38C17.38 10.77 17.99 12 18 14H6z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          20.99
           
          Â°C
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Temperature
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        class="MuiStack-root Card css-1sy57ly-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AirIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14.5 17c0 1.65-1.35 3-3 3s-3-1.35-3-3h2c0 .55.45 1 1 1s1-.45 1-1-.45-1-1-1H2v-2h9.5c1.65 0 3 1.35 3 3zM19 6.5C19 4.57 17.43 3 15.5 3S12 4.57 12 6.5h2c0-.83.67-1.5 1.5-1.5s1.5.67 1.5 1.5S16.33 8 15.5 8H2v2h13.5c1.93 0 3.5-1.57 3.5-3.5zm-.5 4.5H2v2h16.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5v2c1.93 0 3.5-1.57 3.5-3.5S20.43 11 18.5 11z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          0
           
          km/h
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Wind
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        class="MuiStack-root Card css-1sy57ly-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="OpacityIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.66 8 12 2.35 6.34 8C4.78 9.56 4 11.64 4 13.64s.78 4.11 2.34 5.67 3.61 2.35 5.66 2.35 4.1-.79 5.66-2.35S20 15.64 20 13.64 19.22 9.56 17.66 8zM6 14c.01-2 .62-3.27 1.76-4.4L12 5.27l4.24 4.38C17.38 10.77 17.99 12 18 14H6z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          78
           
          %
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Humidity
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        class="MuiStack-root Card css-1sy57ly-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="WaterIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21.98 14H22h-.02zM5.35 13c1.19 0 1.42 1 3.33 1 1.95 0 2.09-1 3.33-1 1.19 0 1.42 1 3.33 1 1.95 0 2.09-1 3.33-1 1.19 0 1.4.98 3.31 1v-2c-1.19 0-1.42-1-3.33-1-1.95 0-2.09 1-3.33 1-1.19 0-1.42-1-3.33-1-1.95 0-2.09 1-3.33 1-1.19 0-1.42-1-3.33-1-1.95 0-2.09 1-3.33 1v2c1.9 0 2.17-1 3.35-1zm13.32 2c-1.95 0-2.09 1-3.33 1-1.19 0-1.42-1-3.33-1-1.95 0-2.1 1-3.34 1-1.24 0-1.38-1-3.33-1-1.95 0-2.1 1-3.34 1v2c1.95 0 2.11-1 3.34-1 1.24 0 1.38 1 3.33 1 1.95 0 2.1-1 3.34-1 1.19 0 1.42 1 3.33 1 1.94 0 2.09-1 3.33-1 1.19 0 1.42 1 3.33 1v-2c-1.24 0-1.38-1-3.33-1zM5.35 9c1.19 0 1.42 1 3.33 1 1.95 0 2.09-1 3.33-1 1.19 0 1.42 1 3.33 1 1.95 0 2.09-1 3.33-1 1.19 0 1.4.98 3.31 1V8c-1.19 0-1.42-1-3.33-1-1.95 0-2.09 1-3.33 1-1.19 0-1.42-1-3.33-1-1.95 0-2.09 1-3.33 1-1.19 0-1.42-1-3.33-1C3.38 7 3.24 8 2 8v2c1.9 0 2.17-1 3.35-1z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          1013
           
          hPa
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Pressure
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        class="MuiStack-root Card css-1sy57ly-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="WbSunnyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          6 : 36 AM
           
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Sun Rise
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        class="MuiStack-root Card css-1sy57ly-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="WbTwilightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m16.9542 8.6615 2.1205-2.122 1.4147 1.4137-2.1205 2.122zM2 18h20v2H2zm9-14h2v3h-2zM3.5426 7.9248l1.4142-1.4142L7.078 8.632l-1.4142 1.4142zM5 16h14c0-3.87-3.13-7-7-7s-7 3.13-7 7z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          5 : 16 PM
           
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Sun Set
        </p>
      </div>
    </div>
  </div>
</div>
`;
